src/Microsoft.SqlTools.ServiceLayer/Management/Common/DataContainer.cs

```diff
@@ -1187,35 +1188,41 @@ internal static System.Security.SecureString BuildSecureStringFromPassword(strin
         private static XmlDocument CreateDataContainerDocument(ConnectionInfo connInfo, bool databaseExists)
         {
             string xml = string.Empty;
+            var serverNameNode = new XElement("servername");
+            serverNameNode.Value = connInfo.ConnectionDetails.ServerName.ToUpper(CultureInfo.InvariantCulture);
 
             if (!databaseExists)
             {
                 xml =
                 string.Format(@"<?xml version=""1.0""?>
                 <formdescription><params>
-                <servername>{0}</servername>
-                <connectionmoniker>{0} (SQLServer, user = {1})</connectionmoniker>
+                {0}
+                <connectionmoniker>{1} (SQLServer, user = {2})</connectionmoniker>
```

Comment:
You shouldn't use the escaped version of the name here - we want the full version here (since it's just a string)

Reply:
I don't see the place where we pass server name to the urn for ObjectUrn like database name, except here, revered the changes and added individual XElement nodes to the other xml elements. Thanks